src/System.CommandLine/Binding/ModelBinder.cs

```diff
@@ -33,9 +33,29 @@ internal ModelBinder(IValueDescriptor valueDescriptor)
 
         public void BindMemberFromValue(PropertyInfo property, IValueDescriptor valueDescriptor)
         {
-            NamedValueSources.Add(
-                (property.PropertyType, property.Name),
-                new SpecificSymbolValueSource(valueDescriptor));
+            var key = (property.PropertyType, property.Name);
+            if (NamedValueSources.TryGetValue(key, out var existingValueSource) &&
+                existingValueSource is null)
+            {
+                // Override existing null value source
+                NamedValueSources[key] = new SpecificSymbolValueSource(valueDescriptor);
+            }
+            else
+            {
+                NamedValueSources.Add(
```

Comment:
It was probably a bug that this called `Add` rather than overwriting, which this change makes clear. We should avoid having the behavior differ based on whether there was a strict binding specified, so let's just always overwrite.